### ISSUE TITLE
LG-286: Allow JSON written to System.out to pass without wrapping

### ISSUE THREAD

**Issue by donbourne** [OpenLiberty/open-liberty#11766]
To address http://www.ibm.com/developerworks/rfe/execute?use_case=viewRfe&CR_ID=132116 

Provide a configuration option which, when enabled, lets JSON written to System.out be written out to the console / messages.log as is when console / message format is json.  Currently all prints/writes to System.out are wrapped in the Liberty's message format JSON, which results in double-wrapped JSON in cases where the print/write is already a JSON formatted string.

When ready, add links to the Upcoming Feature Overview document and Feature Test Summary issue:
- UFO: https://ibm.box.com/s/33q79m28zwl7igyhwzj7x926i6uxzce8
- FTS: #12784 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)

Instructions:
- Do the actions below and mark them complete in the checklist when they are done.
- Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.

## **Design**
Before Development Starts or 8 weeks before Onboarding
- [x] POC Design / UFO Review Scheduled (David Chang) or N/A.
- [x] POC Design / UFO Reviewed (Feature Owner) or N/A.
- [x] Complete any follow-ons from the POC Review.
- [x] Design / UFO Approval (Alasdair Nottingham) or N/A.
- [ ] No Design / No UFO Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A.
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of this epic entitled "FAT Approval Test Summary". Add the link in above.

## **Legal**
3 weeks before Onboarding
- [ ] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.

## **Translation**
3 weeks before Onboarding
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)

## **Feature Complete**
2 weeks before Onboarding
- [x] Implementation complete. (Epic owner / Feature owner)
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [x] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)

## **Focal Point Approvals**
2 to 1 week before Onboarding
#### You **MUST** have the Design Approved or No Design Approved label before requesting focal point approvals.

All features (both "Design Approved" and "No Design Approved")
- [x] **FAT** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [x] **Demo** - (Tom Evans or Chuck Bridgham). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
- [x] **Globalization** (Sam Wong - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.

"Design Approved" features
- [x] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [x] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - (Don Bourne). Serviceability has been addressed.
- [x] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [x] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.

## **Ready for GA**
1 week before Onboarding
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)

## **Other deliverbles**
- [ ] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A.
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
- [ ] **Blog** - (Laura Cowen) Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by mbroz2** [comment]
UFO Forum review comments:
Remove customer names from UFO (slide 8)
Remove (or move them to backup slides section) charts 16, 17, & 18 (seems to have already been done)
Follow up conversation regarding config name for jsonWrapJson vs jsonUnwrapJson (seems to have been already done, with the name now being `appsWriteJson`)

**Comment by Yushan-Lin** [comment]
@gscottj This feature doesn't have UI. May I have accessibility approval?
@gecock SVT is not required for this feature. May I have the SVT approval?
@jhanders34 We don't expect performance hit in this feature. Please assess.
@sabolo  The feature test summary is provided here #12784 
@tevans78 This feature was demoed in 20.13 EOI. May I have the demo approval?

**Comment by gecock** [comment]
@Yushan-Lin I am no longer the SVT focal. Please reach out to @hanczaryk.

**Comment by gscottj** [comment]
Accessibility Verification approved. This feature has no UI and actually makes the text only logs more readable.

**Comment by Yushan-Lin** [comment]
@hanczaryk SVT is not required for this feature. May I have the SVT approval?
@swati do we need STE for this feature?

**Comment by Yushan-Lin** [comment]
@chirp1 ID issue is here: https://github.com/OpenLiberty/docs/issues/1911. May I have ID approval?

**Comment by Yushan-Lin** [comment]
@tevans78 @cbridgha Demo was done in EOI call for 20.13. May I get demo approval?

**Comment by Yushan-Lin** [comment]
@donbourne The following questions has been filled out, may I have serviceability approval?
**1. WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR? Have these issues been addressed in the implementation?**
Yes, the WAD addresses problems customers might encounter. Warning messages were implemented and provides information about issues customers might encounter.

**2. Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).**
a) What problem paths were tested and demonstrated?

User provides an invalid configuration of appsWriteJson.

b) Who did you demo to? Tanya Kulik and Rumana Haque
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?
Yes
**3. SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them. Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.**
a) Who conducted SVT tests for this feature? Tanya Kulik and Rumana Haque
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?
Yes
**4. Which L2 / L3 queues will handle PMRs for this feature? Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it. Ask Don Bourne if you need links or more info.**
Salesforce: WAS L3: Liberty Log Analytics
Retain: LL3ANA, 103

**Comment by Yushan-Lin** [comment]
@skasund I have uploaded the STE slides here https://ibm.ent.box.com/file/691873254646. May I have STE approval?

**Comment by chirp1** [comment]
Approving. The info in the docs issue looks OK. As I discussed with Yushan, the eGA blog needs enough info so that customers can use the new function. The docs might not be updated by eGA as OL writers are working down a prioritized backlog.

**Comment by skasund** [comment]
@Yushan-Lin  Thanks for the STE slides. I've approved the feature.

**Comment by Yushan-Lin** [comment]
@samwatibm translation files have been included, may I have globalization approval?

**Comment by Yushan-Lin** [comment]
@ayoho The feature test summary is provided here #12784 may I have FAT approval?

**Comment by Yushan-Lin** [comment]
@sabolo The feature test summary is provided here #12784 may I have FAT approval?
